lib/tarantool_server.py

```diff
@@ -1270,3 +1276,79 @@ def wait_lsn(self, node_id, lsn):
 
     def get_log(self):
         return TarantoolLog(self.logfile).positioning()
+
+    def get_iproto_port(self, deadline=None):
+        # Check the `box.cfg.listen` option, if it wasn't defined, just return.
+        res = yaml.safe_load(self.admin('box.cfg.listen', silent=True))[0]
+        if res is None:
+            return
+
+        deadline = deadline or 'nil'
+        sleep_time = 0.001
+        timeout = Options().args.server_start_timeout
+
+        version = yaml.safe_load(
+            self.admin("require('tarantool').version", silent=True)
+        )[0]
+        if version.split('.')[0] == '1':
+            # For tarantool 1.x the `box.info.listen` option is not implemented.
+            # This is a simple script to get the server socket of the instance.
+            # At first, it catches both server (listening) and client (sending)
+            # sockets (client sockets usually occur when starting an instance as
+            # a replica). Then it finds the server socket among caught sockets.
+            script = """
+                ffi = require('ffi')
+                socket = require('socket')
+
+                val = ffi.new('int[1]')
+                len = ffi.new('size_t[1]', ffi.sizeof('int'))
+                while not {deadline} or os.time() < {deadline} do
```

Comment:
When `deadline` is None, it'll be `while not None <...>` in Lua. It means accessing to a `None` variable. If it is undefined (it is likely so), this code will fail in the strict mode.

Reply:
I have above `deadline = deadline or 'nil'`.